mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -66,8 +71,25 @@ def message_post(
                         subject = rendered_subject_template + subject
                     elif template.position == "append_after":
                         subject += rendered_subject_template
+                    elif template.position == "inside_replace":
+                        rendered_subject_to_replace = self.env[
+                            "mail.template"
+                        ]._render_template(
+                            template_src=template.subject_to_replace,
+                            model=self._name,
+                            res_ids=[self.id],
+                        )[self.id]
+                        if rendered_subject_to_replace:
+                            # To avoid empty string replacements
+                            subject = subject.replace(
+                                rendered_subject_to_replace,
+                                rendered_subject_template,
+                            )
                 except Exception:
-                    rendered_subject_template = False
+                    _logger.warning(
+                        f"There is an error with {self.display_name} in the mail "
+                        "message custom subject {template.name}"
```

Comment:
issue: there's a bug here (2nd line not rendered). Besides, in general you should use lazy rendering of logger messages, like the rest of Odoo:

```suggestion
                        "There is an error with %s in the mail message custom subject %s",
                        self.display_name,
                        template.name,
```

Reply:
OK about passing the variables as extra arguments of `_` (not sure the syntax though), but not OK about changing to `%s`, and in fact, linter complains about that.

Reply:
The linter only complains when the string is used for translation, because other languages can need to position those terms in a different order. But I'm not telling you to make this string translatable. Logs should be in English always.

By adding the arguments as parameters, you tell Python to don't attempt rendering the message if it's not going to be logged, maybe because the logging level doesn't match your configuration. This can result in better performance. Although indeed the benefit in performance is minimal, it is still the common way to do logging accross all odoo and oca, and it's more comfortable to write and read.

Reply:
Oh, I mixed things in the middle of the discussion and though about translations. Anyway, the f-string syntax is clearer, and I'm sure the performance thing is something that Python itself will improve.

Reply:
OK, then fix the bug:

```suggestion
                        f"There is an error with {self.display_name} in the mail "
                        f"message custom subject {template.name}"
```